dbms/src/Storages/DeltaMerge/File/DMFileReader.cpp

```diff
@@ -142,147 +131,153 @@ bool DMFileReader::getSkippedRows(size_t & skip_rows)
     return next_pack_id < use_packs.size();
 }
 
+// Skip the block which should be returned by next read()
 size_t DMFileReader::skipNextBlock()
 {
-    // Go to next available pack.
-    size_t skip;
-    if (!getSkippedRows(skip))
+    // find the first pack which is used
+    if (size_t skip_rows; !getSkippedRows(skip_rows))
         return 0;
 
-    // Find the next contiguous packs will be read in next read,
-    // let next_pack_id point to the next pack of the contiguous packs.
-    // For example, if we have 10 packs, use_packs is [0, 1, 1, 0, 1, 1, 0, 0, 1, 1],
-    // and now next_pack_id is 1, then we will skip 2 packs(index 1 and 2), and next_pack_id will be 3.
-    const size_t read_pack_limit = read_one_pack_every_time ? 1 : 0;
-    const std::vector<RSResult> & handle_res = pack_filter.getHandleRes();
-    const RSResult expected_handle_res = handle_res[next_pack_id];
-    auto & use_packs = pack_filter.getUsePacks();
+    // move forward next_pack_id and next_row_offset
+    size_t read_rows = getReadRows();
```

Comment:
Compared with the previous implementation, why the `getReadRows()` here can ignore `enable_handle_clean_read == true`?

Reply:
We have separate the logic of clean read for each pack, and we support some packs do clean clean while some packs do not in the same pack.